File: DownloadImage/app/src/main/java/com/example/downloadimage/MainActivity.java
package com.example.downloadimage;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;

import java.io.IOException;
import java.io.InputStream;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.concurrent.ExecutionException;

public class MainActivity extends AppCompatActivity {

    ImageView imageView;

    public void downLoadImage(View view) throws ExecutionException, InterruptedException {
        Log.i("Button","Downloading image");
        ImageDownload task = new ImageDownload();
        Bitmap myImage;

        myImage = task.execute("https://thenypost.files.wordpress.com/2016/09/tv_simpsons_homer1-1a.jpg?quality=80&strip=all&strip=all").get();

        imageView.setImageBitmap(myImage);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        imageView = findViewById(R.id.imageView);

    }

    public class ImageDownload extends AsyncTask<String,Void, Bitmap>{

        @Override
        protected Bitmap doInBackground(String... urls) {

            try {
                URL url = new URL(urls[0]);

                HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();

                urlConnection.connect();

                InputStream io = urlConnection.getInputStream();

                Bitmap myBitmap = BitmapFactory.decodeStream(io);

                return myBitmap;

            } catch (MalformedURLException e) {
                e.printStackTrace();
            } catch (IOException e) {
                e.printStackTrace();
            }
            return null;
        }
    }
}

File: shadeImage/app/src/main/java/com/example/shadeimage/MainActivity.kt
package com.example.shadeimage

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.ImageView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private var bartShading = true;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun fade(view: View){

        val imageView = findViewById<ImageView>(R.id.imageView2);

        val imageView1 = findViewById<ImageView>(R.id.imageViewHomer);

        if (bartShading){

            imageView.animate().alpha(0F).duration = 2000;
            bartShading = false;
            imageView1.animate().alpha(1F).duration = 2000;

        }
        else{
            imageView.animate().alpha(1F).duration = 2000;
//            val imageView1 = findViewById<ImageView>(R.id.imageView2);
            bartShading = true;
            imageView1.animate().alpha(0F).duration = 2000;
        }
//    Log.i("Info","Image view is tapped");


    }

//    fun fadeHomer(view:View){
//        val imageView = findViewById<ImageView>(R.id.imageViewHomer);
//        imageView.animate().alpha(0F).duration = 2000;
//
//        val imageView1 = findViewById<ImageView>(R.id.imageView2);
//        imageView1.animate().alpha(1F).duration = 2000;
//    }
}
